src/python/pants/base/specs.py

```diff
@@ -93,56 +82,15 @@ def matching_addresses(
         :raises: :class:`ResolveError` if no addresses could be matched and this spec type expects
             a match.
         """
-
-
-@dataclass(frozen=True)
-class SingleAddress(AddressSpec):
-    """An AddressSpec for a single address."""
```

Comment:
All these methods were removed because `AddressLiteralSpec` now uses the `AddressInput -> Address` codepath, rather than looking at `AddressFamily`.